mops/include/mops/cuda_opsa.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef MOPS_OPSA_CUDA
+#define MOPS_OPSA_CUDA
+
+#include <cstddef>
+#include <cstdint>
+
+#include "mops/exports.h"
+#include "mops/tensor.hpp"
+
+namespace mops {
+
+namespace cuda {
+
+template <typename scalar_t>
+void outer_product_scatter_add_cuda(
+    const scalar_t *__restrict__ A, // [nedges, nfeatures_A]
```

Comment:
Can we not use the `Tensor<scalar_t, 2>` class here? I though even the CUDA kernel support full C++.